ironfish-rust/src/serializing/mod.rs

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+pub mod aead;
+
+use crate::errors::IronfishError;
+
+/// Helper functions to convert pairing parts to bytes
+///
+/// The traits in the pairing and zcash_primitives libraries
+/// all have functions for serializing, but their interface
+/// can be a bit clunky if you're just working with bytearrays.
+use ff::PrimeField;
+use group::GroupEncoding;
+
+use std::io;
+
+const HEX_CHARS: &[u8; 16] = b"0123456789abcdef";
+
+pub(crate) fn read_scalar<F: PrimeField, R: io::Read>(mut reader: R) -> Result<F, IronfishError> {
+    let mut fr_repr = F::Repr::default();
+    reader.read_exact(fr_repr.as_mut())?;
+
+    Option::from(F::from_repr(fr_repr)).ok_or(IronfishError::InvalidData)
+}
+
+pub(crate) fn read_point<G: GroupEncoding, R: io::Read>(mut reader: R) -> Result<G, IronfishError> {
+    let mut point_repr = G::Repr::default();
+    reader.read_exact(point_repr.as_mut())?;
+
+    Option::from(G::from_bytes(&point_repr)).ok_or(IronfishError::InvalidData)
+}
+
+/// Output the bytes as a hexadecimal String
+pub fn bytes_to_hex(bytes: &[u8]) -> String {
+    let mut hex: Vec<u8> = vec![0; bytes.len() * 2];
+
+    for (i, b) in bytes.iter().enumerate() {
+        hex[i * 2] = HEX_CHARS[(b >> 4) as usize];
+        hex[i * 2 + 1] = HEX_CHARS[(b & 0x0f) as usize];
+    }
+
+    unsafe { String::from_utf8_unchecked(hex) }
+}
```

Comment:
`bytes_to_hex` still uses a `Vec`, due to Rust not providing any useful ways to go between arrays and Strings, which is kind of expected. We could do some mem transmuting ourselves, but no real need, this is _plenty_ fast.

You'll note that there is an `unsafe` block here to call `from_utf8_unchecked` instead of `from_utf8`. It is easy to see that it is impossible to have invalid utf8 since we only parse 4 bits at a time into an array we defined: `HEX_CHARS`.  If someone tried to pass "bad" bytes, they would still need to valid hex.